doc/SAI-IPv6-Segment-Routing-Update.md

```diff
@@ -0,0 +1,484 @@
+
+    /** PSP and USP */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_FLAVOR_PSP_AND_USP,
+
+    /** USD and USP */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_FLAVOR_USD_AND_USP,
+
+    /** PSP and USD */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_FLAVOR_PSP_AND_USD,
+
+    /** PSP, USP and USD */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_FLAVOR_PSP_AND_USP_AND_USD
+
+    /** None */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_FLAVOR_NONE
+
+} sai_local_sid_entry_endpoint_flavor_t;
+```
+
+4. Added a new object sai_local_sid_entry_t configure Endpoint behavior.
+
+```
+/**
+ * @brief Attribute list for Local SID
+ */
+typedef enum _sai_local_sid_entry_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_START,
+
+    /**
+     * @brief Endpoint Function
+     *
+     * @type sai_local_sid_entry_endpoint_type_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE = SAI_LOCAL_SID_ENTRY_ATTR_START,
+
+    /**
+     * @brief Flavor for End, End.X and End.T functions
+     *
+     * @type sai_local_sid_entry_endpoint_flavor_t
+     * @flags CREATE_AND_SET
+     * @default SAI_LOCAL_SID_ENTRY_ENDPOINT_FLAVOR_PSP
+     * @validonly SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_E or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_X or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_T
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_FLAVOR,
+
+    /**
+     * @brief Packet action
+     *
+     * @type sai_packet_action_t
+     * @flags CREATE_AND_SET
+     * @default SAI_PACKET_ACTION_FORWARD
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_PACKET_ACTION,
+
+    /**
+     * @brief Packet priority for trap/log actions
+     *
+     * @type sai_uint8_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_TRAP_PRIORITY,
+
+    /**
+     * @brief Next hop for cross-connect function
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_NEXT_HOP, SAI_OBJECT_TYPE_NEXT_HOP_GROUP, SAI_OBJECT_TYPE_ROUTER_INTERFACE
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     * @validonly SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_X
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_NEXT_HOP_ID,
+
+    /**
+     * @brief VRF for decapsulation and specific table lookup functions
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_VIRTUAL_ROUTER
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     * @validonly SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_T or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT4 or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT6 or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT46
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_VRF,
```

Comment:
While this attr alone works for DT46, how does this work for DT4 or DT6. Don't we need an AFI attr (ipv4/ipv6) along with VRF attr for DT4, DT6?

Reply:
Sorry, I don't understand the issue here. For DT4 or DT6, SAI implementation will check if inner packet type (ipv4 or ipv6) matches with the endpoint_behavior before applying the endpoint functionality. Isn't the AFI implicit with DT4 or DT6?